source/common/network/listen_socket_impl.cc

```diff
@@ -15,6 +15,11 @@ void ListenSocketImpl::doBind() {
     throw EnvoyException(
         fmt::format("cannot bind '{}': {}", local_address_->asString(), strerror(errno)));
   }
+  if (local_address_->ip()->port() == 0) {
```

Comment:
doBind() is also called in UdsListenSocket case so I think this line will crash without checking if ip() is not null or type is IP